cli/remote_build.go

```diff
@@ -156,112 +165,222 @@ func (p *RemoteCmd) Execute(_ context.Context, f *flag.FlagSet, _ ...interface{}
+				if err != nil {
+					fmt.Printf("Unable to move %s -> %s in the temporary cloned repository: %v\n", s.Extra, n, err)
+					return subcommands.ExitFailure
+				}
+			} else {
+				// Copy contents from the workRepo fs to cloneRepo fs
+				originalFile, err := worktree.Filesystem.Open(n)
+				if err != nil {
+					fmt.Printf("Unable to open %s on the work tree: %v\n", n, err)
+					return subcommands.ExitFailure
+				}
+
+				newFile, err := clonedWorktree.Filesystem.Create(n)
+				if err != nil {
+					fmt.Printf("Unable to open %s on the cloned tree: %v\n", n, err)
+					return subcommands.ExitFailure
+				}
+
+				_, err = io.Copy(newFile, originalFile)
+				if err != nil {
+					fmt.Printf("Unable to copy %s: %v\n", n, err)
+					return subcommands.ExitFailure
+				}
+				_ = originalFile.Close()
+				_ = newFile.Close()
+
+				// Add each detected changed file to the clonedRepo index
+				_, err = clonedWorktree.Add(n)
+				if err != nil {
+					fmt.Printf("Unable to add %s to the temporary cloned repository: %v\n", n, err)
+					return subcommands.ExitFailure
+				}
+			}
+		}
+
+		latest, err = commitTempChanges(clonedWorktree)
+		if err != nil {
+			fmt.Printf("Commit to temporary cloned repository failed: %v\n", err)
+			return subcommands.ExitFailure
+		}
+
+		baseCommit, err = clonedRepo.CommitObject(commonAncestor)
+		if err != nil {
+			fmt.Printf("Commit definition failed: %v\n", err)
+			return subcommands.ExitFailure
+		}
+
+		tempCommit, err := clonedRepo.CommitObject(latest)
+		if err != nil {
+			fmt.Printf("Commit definition failed: %v\n", err)
+			return subcommands.ExitFailure
+		}
+
+		fmt.Printf("Generating a patch from comparing %s to %s\n", baseCommit.Hash, tempCommit.Hash)
+
+		patch, err = baseCommit.Patch(tempCommit)
+		if err != nil {
+			fmt.Printf("Patch generation failed: %v\n", err)
+			return subcommands.ExitFailure
+		}
+
+		//p.patchData = append(p.patchData, []byte(patch.String())...)
+		p.patchData = []byte(patch.String())
+
+	}
 
 	if p.patchPath != "" {
 		if err := savePatch(p); err != nil {
 			fmt.Printf("\nUnable to save copy of generated patch: %v\n\n", err)
 		} else {
-			fmt.Printf("Patch copy saved at: %v\n", p.patchPath)
+			fmt.Printf("Patch saved at: %v\n", p.patchPath)
 		}
 	}
 
-	err = submitBuild(project, p, target.Tags)
+	if !p.dryRun {
+		fmt.Println("Submitting Remote Build ...")
+		err = submitBuild(project, p, target.Tags)
 
-	if err != nil {
-		fmt.Printf("Unable to submit build: %v\n", err)
-		return subcommands.ExitFailure
+		if err != nil {
+			fmt.Printf("Unable to submit build: %v\n", err)
+			return subcommands.ExitFailure
+		}
+	} else {
+		fmt.Println("Dry run ended, build not submitted")
 	}
 
 	return subcommands.ExitSuccess
 }
 
+func commitTempChanges(w *git.Worktree) (latest plumbing.Hash, err error) {
+	latest, err = w.Commit(
+		"YourBase remote build",
+		&git.CommitOptions{
+			Author: &object.Signature{
+				Name:  "YourBase",
+				Email: "robot@yourbase.io",
```

Comment:
We should make a TODO to get this from their config or from the API so that people know who's building